addon/components/inputs/datetime.js

```diff
@@ -13,12 +14,40 @@ export default AbstractInput.extend({
 
   layout,
 
+  // == State Properties =======================================================
+
+  getDefaultProps () {
+    if (this.get('value')) {
+      var arr = this.get('value').split('T')
+
+      return {
+        date: arr[0] || '',
+        time: arr[1] || ''
+      }
+    }
+
+    return {
+      date: '',
+      time: ''
+    }
+  },
```

Comment:
Rather than maintaining state for the date and time I'd prefer to see you just use computed properties.

```js
/**
 * Get formatted datetime in a specific format
 * @param {String} value - datetime as a string
 * @param {String} format - format to represent datetime as
 * @returns {String} datetime in desired format
 */
function getFormattedDate (value, format) {
  if (typeOf(value) !== 'string') {
    return ''
  }

  const time = moment(value).format(format)
  return time === 'Invalid date' ? '' : time
}

export default AbstractInput.extend({

  …

  @readOnly
  @computed('value')
  date (value) {
    return getFormattedDate(value, 'YYYY:MM:DD')
  },

  @readOnly
  @computed('value')
  time (value) {
    return getFormattedDate(value, 'HH:mm:ss')
  }
```

Then in your template you can pass `date` and `time` directly into the two components.